browser/models.py

```diff
@@ -156,166 +222,248 @@ def create(filename, input_bucket, output_bucket, path, rgb=False):
             input_bucket (str): S3 bucket to download file
             output_bucket (str): S3 bucket to upload file
             path (str): full path to download & upload file in buckets; includes filename
-            rgb (bool): whether to display raw frames in RGB mode
 
         Returns:
             Project: new row in the Project table
         """
         start = timeit.default_timer()
-        new_project = Project(filename, input_bucket, output_bucket, path, rgb=rgb)
+        new_project = Project(filename, input_bucket, output_bucket, path)
         db.session.add(new_project)
         db.session.commit()
-        current_app.logger.debug('Created new project with ID = "%s" in %ss.',
-                                 new_project.id, timeit.default_timer() - start)
+        new_project.create_memento('create_project', all_frames=True)
+        db.session.commit()
+        logger.debug('Created new project %s in %ss.',
+                     new_project.id, timeit.default_timer() - start)
         return new_project
 
     def update(self):
         """
         Commit the project changes from an action.
+        Records the effects of the action in the Action table.
         """
         start = timeit.default_timer()
+        if self.action.labels_changed:
+            self.labels.update()
         db.session.commit()
         logger.debug('Updated project %s in %ss.',
                      self.id, timeit.default_timer() - start)
 
     def finish(self):
```

Comment:
do rows in the action table need to get cleared upon project finish?

Reply:
We don't clear them out currently, as Erick & Dave mentioned interest in building an "action dataset" for reinforcement learning/other uses. As we only stored the changed frames in the action history, this should be relatively lightweight, but we should monitor the database size to see if this become unwieldy.